craft_parts/parts.py

```diff
@@ -177,7 +177,9 @@ def get_scriptlet(self, step: Step) -> Optional[str]:
         return self._data.get(scr[step])
 
 
-def part_list_by_name(names: Optional[List[str]], part_list: List[Part]) -> List[Part]:
+def part_list_by_name(
+    names: Optional[Sequence[str]], part_list: List[Part]
```

Comment:
probably want part_list to be a Sequence as well

Reply:
Will adjust that in a follow-up, I want to check all the consequences in the rest of the code.